test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -283,7 +298,11 @@ protected void configure() {
                         bind(NamedWriteableRegistry.class).toInstance(namedWriteableRegistry);
                     }
                 }
-        ).createInjector();
+        );
+        pluginsService.processModules(modulesBuilder);
+        injector = modulesBuilder.createInjector();
+        IndexScopedSettings indexScopedSettings = injector.getInstance(IndexScopedSettings.class);
+        idxSettings = IndexSettingsModule.newIndexSettings(index, indexSettings, indexScopedSettings);
```

Comment:
why is this moved down here and why do you need to Initialize `idxSettings` so late?